### ISSUE TITLE
Add validation rules in useForm hook ?

### ISSUE THREAD

**Issue by raskyer** [react-hook-form/react-hook-form#10058]
**Is your feature request related to a problem? Please describe.**
In my application, its sometimes kinda hard to follow along all the front end rules on my forms. Mainly because I use <Controller> with React MUI and it tends to make the component really verbose.

**Describe the solution you'd like**
I would like to be able to describe rules of an input directly in useForm hook.

**Describe alternatives you've considered**
A clear alternative is to use zod or yup (with their resolvers) but it means I have to import this library in the front end and I want to reduce the bundle size.

**Additional context**
```
const form = useForm({
  defaultValues: {
    field: ''
  },
  rules: {
    field: {
      required: true,
      ...etc
    }
  }
});
```

**Comment by bluebill1049** [comment]
have you considered using a resolver such as zod or yup? They are great at the form level.

**Comment by raskyer** [comment]
Yes, as mention in my ticket 

"A clear alternative is to use zod or yup (with their resolvers) but it means I have to import this library in the front end and I want to reduce the bundle size."
